### ISSUE TITLE
Setting to limit local organisation's visibility only to their own organisation

### ISSUE THREAD

**Issue by tking2** [MISP/MISP#2442]
I'm looking for some way to limit a Local Organisation's visibility only to their own organisation, so they are not able to enumerate other Orgs on the same MISP instance at https://<misp_url>/organisations/index

This would also apply to other organisations defined in the same sharing group as them on sharing_groups/index.

We have orgs who require access to our MISP to sync, but we don't want other orgs to see each other.

If this could be a global setting in MISP, disabling viewing other orgs unless you are a Site Admin.

**Comment by iglocska** [comment]
Just had a chat with Foating**Code**[Andras is a big meanie] about this today, sounds a bit tricky but doable. The issues I see currently:

1. The sharing group information is available via both the UI and the api, getting the latter is trivial for users
2. the API way of retrieving that information is the same functionality used to pull events from an instance to another

Bottom line, we need to figure out a clean way to enforce this for non-sync users only, it's definitely doable but it will take a bit more time and has to come after the objects.

If there are any other things I might be missing here that you guys need, just put them in the issue here and I'll keep it in mind!

**Comment by tking2** [comment]
Makes sense, our sharing groups may be different in the sense that we only release to Local Organisations who we've defined in our MISP - This allows them to setup a sync to receive events within the sharing group to only their local instance. No more hops are possible. With this approach, the receiving org doesn't need to know the other organisations defined in the group, as they do not need to pass the data onto anyone else. 

Looking at the JSON of an event, the Sharing Group is defined as:
```
        "SharingGroup": {
            "id": "1",
            "name": "SharingGroupName",
            "releasability": "Releasable Text",
            "description": "Sharing Group",
            "uuid": "<uuid>",
            "organisation_uuid": "<org_uuid>",
            "org_id": "1",
            "sync_user_id": "0",
            "active": true,
            "created": "",
            "modified": "",
            "local": true,
            "roaming": false
        }
```
This data being released in this fine - It doesn't define other local organisations who have access to the same sharing group. However viewing the sharing group directly at <misp_url>/sharing_groups/index allows an org to see all Organisations defined. Only orgs that have the capability to extend the sharing group would require to see all orgs to allow them to extend?

As for viewing Organisations - Is there a requirement for organisations to view other organisations on the same instance by going to /organisations/index (if the user trying to view isn't a Site Admin)? Obviously if Organisations on the MISP create an event which they can see, then that's fair enough, as it'll appear under the Event Org. Just any way to disable the viewing this page is all we need.

**Comment by iglocska** [comment]
Oh wow, you are correct, I must have implemented some sanitisation for the sharing group in the JSON for this reason already, didn't remember doing that. Site admins and sync users see the full thing. So it's an even simpler task than I thought.

Sounds good, should be doable.

**Comment by tomking2** [comment]
Hey Andras,
Don't suppose you've had a chance to look at this one yet?

**Comment by adulau** [comment]
@iglocska Maybe this could be useful for X-ISAC too?

**Comment by iglocska** [comment]
Semi implemented now. There is a new setting that allows you to hide the org index and all relevant statistics pages from normal users. However, they can still see organisations that have created data and organisations within sharing groups (the latter we still need to look into).

One massive caveat: Sharing group editors and site admins are exempt from the restriction (as you can't create a sharing group without being able to see the list of orgs to add them to the sharing group).
